Automattic/jetpack
# Stats: Make the Stats dashboard widget link to Odyssey stats if enabled

## Proposed changes:
<img width="537" alt="image" src="https://user-images.githubusercontent.com/746152/217671270-a2e1671d-51cd-4788-ad98-98ee048c9d53.png">

* Makes the **View all stats** link point to the wp-admin stats page if Odyssey Stats is enabled.

|Before|After|
|-|-|
|<img width="1054" alt="image" src="https://user-images.githubusercontent.com/4044428/217680641-a5b08668-b41f-47f7-9f41-c04076235c18.png">|<img width="1081" alt="image" src="https://user-images.githubusercontent.com/4044428/217680550-59699768-6064-4c00-8824-42d116d006b1.png">|

* Hides the "View on WordPress.com" link under the Jetpack Stats section in the At-a-Glance section when Odyssey Stats is enabled.

### Other information:


## Jetpack product discussion


## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
1. Check out this branch.
1. Enable Odyssey stats.
3. Visit the WordPress Dashboard (`/wp-admin/`)
4. Find the Stats widget and click the **View all stats** link. Ensure it points to `/wp-admin/admin.php?page=stats`.
5. Visit Jetpack's At-a-Glance Dashboard (`/wp-admin/admin.php?page=jetpack#/dashboard`).
6. Ensure that you only see the "View Detailed Stats" button below the Jetpack Stats chart. "View on WordPress.com" should not be present.
